library/accessory_workflows/subset_fastq_dataset/subset_fastq_dataset.wdl

```diff
@@ -0,0 +1,149 @@
+workflow SubsetFastqDataset {
+    Array[File] bams
+    Array[File] fastqs
+    String keepregion
+
+    call MergeBams {
+        input:
+        aligned_bams = bams
+    }
+
+    call IndexSortConcatenated {
+        input:
+        concat_bam = MergeBams.output_concat_bam
+    }
+
+    call ExtractReadNames {
+        input:
+        sorted_bam = IndexSortConcatenated.output_sorted_bam,
+        sorted_bai = IndexSortConcatenated.output_sorted_bai,
+        subset_region = keepregion
+    }
+
+    scatter (fastq in fastqs) {
+        call FilterFastq {
+            input:
+            fastq = fastq,
+            keep_read_names_gz = ExtractReadNames.output_kept_reads
+        }
+    }
+
+    call TarFilteredFastq {
+        input:
+        filtered_fastqs = FilterFastq.output_filtered_fastq_gz
+    }
+
+    output {
+        File output_tar = TarFilteredFastq.output_tarred_fastqs
```

Comment:
How come the output of this workflow is a tarred file? An `Array[File]` would mean that we no longer have to include `pigz` in the docker image and get rid of the untarring steps in tasks. It also means that the output files can be looked at individually instead of having to download an entire dataset. Unless this is a requirement for the WDL, I would suggest that we don't compress files in the WDL to keep things as easy to comprehend as possible.

Reply:
Can be improved, but haven't worked with Arrays in WDL before so not confident to fix this.